### PR TITLE
Updated dependencies, added eslint, rollup, plus some fixes

### PR DESCRIPTION
**Pull request by taoeffect**
This PR

- Removes the no-longer-maintained `vueify` and switches us from `browserify` to the more advanced (and more rigid) `rollup`
- Removes all related dependencies to vueify and browserify
- Updates many dependencies (including babel to version 7) and fixes critical npm audit warnings
- Extracts component scss out of the bundle and into a `component.css` file
- Adds support for dynamically lazy-loading components with `import()` and shows an example of how to do that with `UserGroup.vue` in `router.js`
- Moved all files in `frontend/simple` up one directory and placed all router links under `/app` URL (#526) in order to make it easier to serve the app using third-party tools
- Significantly improves our linter by switching from `standard` to `eslint` with `eslint-plugin-standard`
- Deleted all the old stuff in `frontend/_static`
- Creates a new top-level folder called `historical` under which old but potentially useful code can be kept
- Updates `browserslist` key in `package.json` so that `async/await` do not get transformed by babel

I recommend viewing the diff with whitespace changes [turned off](https://github.com/okTurtles/group-income-simple/pull/521/files?utf8=✓&diff=split&w=1).

### TODO

- [x] fix import of circle slider
- [x] Try Grunt integration with `rollup.watch` instead of `rollup.rollup`! ~~Replace `rollup-plugin-serve` with [`serve`](https://github.com/zeit/serve-handler#rewrites-array), or better yet, ~~just keep what we currently have with grunt and run rollup via grunt / grunt-exec~~ that didn't work out too well. So now am thinking maybe I should try with [browser-sync](https://github.com/Browsersync/browser-sync)? Or maybe just switch to webpack...? Maybe leave it all for a future PR.~~
- [x] ~~Remove all browserify related dependencies and migrate any remaining stuff to rollup (for example, by using `script2ify` with `rollup-plugin-browserify-transform`~~
- [x] See if it is better to replace Grunt with npm scripts or to stick with it
- [x] Use eslint with `eslint-config-standard` so that it lints `.vue` files (instead of using standard directly, because it doesn't support that for some reason)
- [x] Make the SASS setup coherenet, simple and DRY
- [x] Connect the SASS setup with the vue code so that JS has access to theme variables from the SASS
- [ ] *NOTE: this is now part of issue #525* - Verify that (1) vueify is loading the sass that's imported in `Sidebar.vue`, and (2) that when I removed `../../` from `// @import "../../node_modules/bulma/sass/utilities/initial-variables";` (and all the other `../../node_modules`), that I didn't break anything. Re-add the `../../` if I did. Choose either the `grunt-sass`, or better yet (so that `@import` can be called from components), remove `grunt-sass` and use the bundler's sass plugin (whatever the equivalent for vueify/rollup/webpack is)
- [x] Create issues out of any of the above that need to be created, as well as the rollup log output below:

```
(!) VuePlugin plugin: <contribution v-for="contribution in fakeStore.receiving.nonMonetary">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/Contributions.vue
(!) VuePlugin plugin: <contribution v-for="contribution in fakeStore.giving.nonMonetary">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/Contributions.vue
(!) VuePlugin plugin: <voting-banner v-for="proposal in Object.values(proposals)">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/containers/Proposals.vue
(!) VuePlugin plugin: <voting v-for="proposal in groupProposals.notVoted">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/containers/Proposals.vue
(!) VuePlugin plugin: <voting v-for="proposal in groupProposals.own">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/containers/Proposals.vue
(!) VuePlugin plugin: <voting v-for="proposal in groupProposals.alreadyVoted">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/containers/Proposals.vue
(!) VuePlugin plugin: <list-item v-for="id in list.order">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/components/Chatroom/ConversationsList.vue
(!) VuePlugin plugin: <menu-item v-for="group in groupsByName">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/containers/sidebar/GroupsList.vue
(!) VuePlugin plugin: <message v-for="message in ephemeral.pendingMessages">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/components/Chatroom/ChatMain.vue
(!) VuePlugin plugin: <avatar v-for="user in founders">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
frontend/simple/views/containers/Chatroom/ConversationGreetings.vue
```

### CODE REVIEW

**Comment by taoeffect** [comment]
OK @pieer this PR is ready for your review! 😄 🎉 

Please also see the related issue that I just created: #525

File: frontend/main.js [hunk 1]
```diff
@@ -53,6 +53,8 @@ async function startApp () {
   new Vue({
     router: router,
     components: {
+      // Sidebar: await import('./views/containers/sidebar/Sidebar.vue'),
```
Comment (pieer) [inline frontend/main.js]: Remove commented code?

File: frontend/views/CreateGroup.vue [hunk 2]
```diff
@@ -78,21 +78,24 @@
   .fade-leave-to /* .fade-leave-active below version 2.1.8 */ {
     opacity: 0;
   }
-</style>
-<style scoped>
+
```
Comment (pieer) [inline frontend/views/CreateGroup.vue:83]: Any reason why this code is going to global css now?

Comment (taoeffect) [reply frontend/views/CreateGroup.vue:83]: I will add the scoped back in, but also have added a bunch of comments about doing this properly with `transitions.js` (which currently has a poor implementation that isn't really being used), that's something that I hope you can improve.

Comment (taoeffect) [reply frontend/views/CreateGroup.vue:83]: Created #529 to address this.

**Review by pieer — COMMENTED** [review]
All good. Great work!

**Review by pieer — APPROVED** [review]

**Pull request merged**